internal/chezmoi/findexecutable.go

```diff
@@ -0,0 +1,47 @@
+package chezmoi
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	foundExecutableCacheMutex sync.Mutex
+	foundExecutableCache      = make(map[string]struct{})
+)
+
+// FindExecutable is like lookPath except that you can specify the paths rather than just using the current `$PATH`. This
+// makes it useful for the resulting path of rc/profile files.
+func FindExecutable(file string, paths ...string) (string, error) {
+	foundExecutableCacheMutex.Lock()
+	defer foundExecutableCacheMutex.Unlock()
+
+	// stolen from: /usr/lib/go-1.20/src/os/exec/lp_unix.go:52
+	for _, dir := range paths {
+		if dir == "" {
+			continue
+		}
+		p := filepath.Join(dir, file)
+		for _, path := range findExecutableExtensions(p) {
```

Comment:
Change this to:

```go
for _, f := range candidates {
  path := filePath.Join(dir, f)
```

As far as I can tell, we can’t avoid the double loop, but we *can* reduce the number of internal loops by not calling `findExecutableExtensions` for each path entry.

Reply:
As per: https://github.com/twpayne/chezmoi/pull/3162#discussion_r1287882105 You are correct since the path doesn't matter. Will address on update.

Not sure why I didn't catch it.